plugins/helm/parser/src/templateanalyzer.h

```diff
@@ -0,0 +1,140 @@
+#ifndef CC_PARSER_TEMPLATEANALYZER_H
+#define CC_PARSER_TEMPLATEANALYZER_H
+
+#include "yaml-cpp/yaml.h"
+
+#include "model/file.h"
+
+#include <model/microservice.h>
+#include <model/microservice-odb.hxx>
+#include <model/microserviceedge.h>
+#include <model/microserviceedge-odb.hxx>
+#include <model/msresource.h>
+#include <model/msresource-odb.hxx>
+#include <model/helmtemplate.h>
+#include <model/helmtemplate-odb.hxx>
+
+#include <parser/parsercontext.h>
+
+namespace cc
+{
+namespace parser
+{
+
+class TemplateAnalyzer
+{
+public:
+  TemplateAnalyzer(
+    ParserContext& ctx_,
+    std::map<std::string, YAML::Node>& fileAstCache_);
+
+  ~TemplateAnalyzer();
+
+  void init();
+  uint64_t getTemplateCounter() { return templateCounter; }
+
+private:
+  bool visitKeyValuePairs(
+    std::string path_,
```

Comment:
Code smell: string parameter passed by value, instead of const reference. Is it required to make a copy here?